### PR TITLE
chore: update cypress and happo

### PR DESCRIPTION
**Pull request by TomasSlama**
### Description

Upgrade cypress and happo to the latest version

### How to test

- CI should be green

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

### CODE REVIEW

**Review by OndrejTuma — APPROVED** [review]
👍

File: package.json [hunk 1]
```diff
@@ -23,10 +23,10 @@
     "syncpack:fix": "syncpack fix-mismatches",
     "symlink": "cross-env ./bin/symlink --link",
     "symlink:off": "cross-env ./bin/symlink --unlink",
-    "test": "yarn test:unit && yarn test:cypress",
-    "test:cypress": "yarn build:package && yarn test:setup cypress run-ct",
-    "test:cypress:ci": "happo-cypress -- cypress run-ct",
-    "test:cypress:open": "yarn build:package && yarn test:setup cypress open-ct",
+    "test": "yarn test:unit && yarn test:integration",
+    "test:integration": "yarn build:package && yarn test:setup cypress run-ct",
+    "test:integration:ci": "yarn happo-e2e -- -- yarn cypress run-ct",
```
[the remark this answers is not in the document]
Comment (TomasSlama) [reply package.json:28]: @denieler detailed info in [changelog](https://github.com/happo/happo-cypress/releases/tag/v3.0.0), but to simplify, the happo wants to support different libraries for E2E, so they extracted common parts to `happo-e2e`

File: package.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
     "generate:example": "hygen component example",
     "generate:icons": "./bin/generate-icons",
     "happo": "cross-env TEST_ENV=visual happo run",
-    "happo:ci": "cross-env TEST_ENV=visual happo-ci-github-actions",
+    "happo:ci": "cross-env TEST_ENV=visual HAPPO_IS_ASYNC=false happo-ci-github-actions",
```
[the remark this answers is not in the document]
Comment (TomasSlama) [reply package.json:17]: [docs](HAPPO_IS_ASYNC=false), When setting `HAPPO_IS_ASYNC` is true, CI job will not wait for the report to be created.

**Review by denieler — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -36,8 +36,8 @@ In order to run storybook you need to execute `yarn start` which will spin up st
 | **lint**               | Lint all files                                                                 |
 | **start**              | Start storybook instance and inspect components                                |
 | **test**               | Run jest and cypress tests                                                     |
-| **test:cypress**       | Run cypress tests                                                              |
-| **test:cypress:open**  | Run cypress in development mode                                                |
+| **test:integration**       | Run cypress tests                                                              |
+| **test:integration:open**  | Run cypress in development mode                                                |
```
Comment (separatio) [inline README.md:40]: You should probably indent the rest of the lines properly.

**Review by separatio — APPROVED** [review]
I got one nit in here, but it looks good overall. Didn't test.

File: .github/workflows/pr.yaml [hunk 4]
```diff
@@ -56,10 +56,6 @@ jobs:
     env:
       NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
 
-    container:
```
[the remark this answers is not in the document]
Comment (dmaklygin) [reply .github/workflows/pr.yaml:59]: wdyt if we reuse custom runners?

```
runs-on: ['org/toptal', 'os/linux', 'arch/x64', 'size/large']
```

File: README.md (hunk 3 above)
Comment (dmaklygin) [inline README.md:40]:
```suggestion
| **test:integration**       | Run integration tests                                                              |
| **test:integration:open**  | Run integration in development mode                                                |
```

File: package.json [hunk 5]
```diff
@@ -83,9 +83,9 @@
     "copy-to-clipboard": "^3.3.1",
     "core-js": "^3.6.5",
     "cross-env": "^7.0.3",
-    "cypress": "^9.5.0",
+    "cypress": "^9.5.4",
```
Comment (dmaklygin) [inline package.json:86]: I would reuse `davinci-qa` package for conducting cypress tests. have you checked the possibility of reusing `davinci-qa` ?

Comment (TomasSlama) [reply package.json:86]: definitely a good idea, I wanted to do it in separate PR. Even for unit tests

File: README.md (hunk 3 above)
Comment (TomasSlama) [reply README.md]: @separatio ✅

Comment (TomasSlama) [reply README.md]: @dmaklygin ✅

**Review by dmaklygin — APPROVED** [review]

**Review by dmaklygin — APPROVED** [review]

**Pull request merged**